fairroot/geobase/FairGeoSet.cxx

```diff
@@ -517,3 +519,56 @@ void FairGeoSet::compare(FairGeoSet& rset)
     }
     cout << "Number of additional volumes in second list: " << nnf << '\n';
 }
+
+Int_t FairGeoSet::getModNumInMod(const TString& name)
+{
+    // check if the naming is correct. All modules in a geometry file must start with
+    // the generic module name followed by a number.
+    if ((!name.BeginsWith(GetName())) || (name.Length() <= fName.Length())) {
+        LOG(fatal) << "Naming conventions violated. The element name must start with " << GetName()
+                   << " followed by a unique number.\n"
+                   << "The passed value was " << name;
+    }
+    // construct TString starting after the end of the detector name
+    // which contains the detector number
+    TString modNumber(name(fName.Length(), (name.Length() - fName.Length())));
+
+    // check if the conversion worked as expected. Atoi internally calls the c function
+    // atoi which returns in case of failure the value 0. Since in our case a value of
+    // 0 is an error anyway (Natural counting starts at 1 as well as the counting of elements
+    // in the module) a value of 0 indicates an error
+    int retVal = modNumber.Atoi();
+    if (0 == retVal) {
+        LOG(fatal) << "Naming conventions violated. The element name must start with " << GetName()
```

Comment:
```suggestion
    const auto& prefix = fName;
    const auto prefixLength = prefix.Length();
    const auto length = name.Length();
    // check if the naming is correct. All modules in a geometry file must start with
    // the generic module name followed by a number.
    if ((!name.BeginsWith(prefix)) || (length <= prefixLength)) {
        LOG(fatal) << "Naming conventions violated. The element name must start with " << prefix
                   << " followed by a unique number.\n"
                   << "The passed value was " << name;
    }
    // construct TString starting after the end of the detector name
    // which contains the detector number
    TString modNumber(name(prefixLength, (length - prefixLength)));

    // check if the conversion worked as expected. Atoi internally calls the c function
    // atoi which returns in case of failure the value 0. Since in our case a value of
    // 0 is an error anyway (Natural counting starts at 1 as well as the counting of elements
    // in the module) a value of 0 indicates an error
    int retVal = modNumber.Atoi();
    if (0 == retVal) {
        LOG(fatal) << "Naming conventions violated. The element name must start with " << prefix
```

Yes, in optimized builds all those function calls are probably inlined because they are defined in header files, but I think the variables state a much clearer intent which also improves readability, no?

Reply:
Changed. I agree that the suggestion improved the readability.